target/linux/ath79/dts/qca9557_dongwon_dw02-412h.dtsi

```diff
@@ -0,0 +1,165 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qca955x.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	aliases {
+		led-boot = &led_wan;
+		led-failsafe = &led_wan;
+		led-upgrade = &led_wan;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		wps {
+			label = "WPS button";
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+		
+		reset {
+			label = "Reset button";
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_wan: wan {
+			label = "green:wan";
+			gpios = <&gpio 22 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan {
+			label = "green:wlan";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&spi {
+	status = "okay";
+
+	num-cs = <1>;
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <50000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x040000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x040000 0x010000>;
+			};
+
+			partition@50000 {
+				label = "log";
+				reg = <0x050000 0x010000>;
+				read-only;
+			};
+
+			partition@60000 {
+				label = "recoverk";
+				reg = <0x060000 0x0e0000>;
+				read-only;
+			};
+
+			partition@140000 {
+				label = "recoverr";
+				reg = <0x140000 0x090000>;
+				read-only;
+			};
+
+			partition@1d0000 {
+				label = "nvram";
+				reg = <0x1d0000 0x010000>;
+				read-only;
+			};
+
+			partition@1e0000 {
+				label = "nvbackup";
+				reg = <0x1e0000 0x010000>;
+				read-only;
+			};
+
+			art: partition@1f0000 {
+				label = "art";
+				reg = <0x1f0000 0x010000>;
+				read-only;
+			};
+		};
+	};
+};
+
+&pcie0 {
+	status = "okay";
+
+	wifi@0,0 {
+		compatible = "qcom,ath10k";
+		reg = <0 0 0 0 0>;
+	};
+};
+
+&wmac {
+	status = "okay";
+
+	mtd-cal-data = <&art 0x1000>;
+	mtd-mac-address = <&art 0x0>;
+	mtd-mac-address-increment = <3>;
+};
+
+&usb_phy0 {
+	status = "okay";
+};
+
+&usb0 {
+	status = "okay";
+};
+
+&mdio0 {
+	status = "okay";
+
+	phy0: ethernet-phy@0 {
+		reg = <0>;
+
+		qca,ar8327-initvals = <
+			0x04 0x07600000 /* PORT0 PAD MODE CTRL */
+			0x50 0xcf37cf37 /* LED Control Register 0 */
+			0x54 0x00000000 /* LED Control Register 1 */
+			0x58 0x00000000 /* LED Control Register 2 */
+			0x5c 0x0030c300 /* LED Control Register 3 */
+			0x7c 0x0000007e	 /* PORT0_STATUS */
+			>;
+	};
+};
+
+&eth0 {
+	status = "okay";
+
+	mtd-mac-address = <&art 0x0>;
+	mtd-mac-address-increment = <1>;
```

Comment:
From what I remember,
```
   lan: label +2
   wan: label +0/+1
wlan5g: label +4
wlan2g: label +3
```
Could you check the LAN address again?

Reply:
label: 9D

lan: 9E (+1)
wan: 9D (+0)
wlan5g: A1 (+4)
wlan2g: A0 (+3)

I set IP allocation policy in stock firmware to router mode, not KT mode.

Reply:
Hmm okay, I haven't checked the mode on the latest stock FW so probably you are right.

Reply:
If you checked, please document that in the commit message. While at it, please wrap the commit message to a maximum of 75 characters.

Reply:
I didn't realize that I had to keep this... Thanks for informing me about this!